apps/checkout/src/components/Radio.tsx

```diff
@@ -1,29 +1,49 @@
-import React, { cloneElement, ReactNode } from "react";
+import React, { ReactNode } from "react";
 import { Text } from "@components/Text";
 import clsx from "clsx";
 
-export interface RadioOption {
+export interface RadioOptionBase {
   onSelect: (value: string) => void;
   selectedValue: string | undefined;
   value: string;
-  title?: string;
-  subtitle?: string;
-  content?: ReactNode;
   disabled?: boolean;
 }
 
-export interface RadioOptionContentProps {
+export type ContentFunction = (
+  props: Pick<RadioOptionContentProps, "htmlFor">
+) => ReactNode;
+
+export interface CustomRadioOption extends RadioOptionBase {
+  content: ContentFunction;
+  title?: never;
+  subtitle?: never;
+}
+
+export interface SimpleRadioOption extends RadioOptionBase {
+  title: string;
+  subtitle?: string;
+  content?: never;
+}
+
+export type RadioOption = SimpleRadioOption | CustomRadioOption;
```

Comment:
I think this is overly complex in usage, what if we just use `children` and type it as: `string | ReactNode`. In code we can do a simple check:
```
typeof children === "string" ? <label>....</label> : children
```
The end user shouldn't have to check the implementation to see what's supposed to be used and not everyone might use TS to easily catch if he's misusing it.
```

Reply:
In case someone doesn't use typescript, they'll have to look at the code anyway to know what props they should pass. But using children seems a good idea, will change 👍